File: webp.js
const sharp = require("sharp");
const fs = require("fs");
const path = require("path");

function convert_to_webp(input_dir, output_dir, targetSizeKB = 80) {
	if (!fs.existsSync(output_dir)) {
		fs.mkdirSync(output_dir);
	}

	fs.readdirSync(input_dir).forEach((file) => {
		const ext = path.extname(file);
		if (ext === ".jpg" || ext === ".png") {
			const img = sharp(path.join(input_dir, file));
			const webp_filename = path.join(
				output_dir,
				path.basename(file, ext) + ".webp"
			);

			img.toBuffer()
				.then((data) => {
					let quality = Math.round(
						(targetSizeKB * 1024 * 100) / data.length
					);
					quality = Math.min(100, quality); // Quality cannot exceed 100
					return sharp(data)
						.webp({ quality: quality })
						.toFile(webp_filename);
				})
				.catch((err) => console.error(err));
		}
	});
}

// Usage
convert_to_webp("input-dir", "output-dir", 80);

File: webp.py
from PIL import Image
import os

def convert_to_webp(input_dir, output_dir, quality=80):
    if not os.path.exists(output_dir):
        os.makedirs(output_dir)

    for filename in os.listdir(input_dir):
        if filename.endswith(".jpg"):
            img = Image.open(os.path.join(input_dir, filename))
            webp_filename = os.path.splitext(filename)[0] + ".webp"
            img.save(os.path.join(output_dir, webp_filename), "webp", quality=quality)

# Usage
convert_to_webp('images/jpg', 'jpg-img', quality=80)
